article/app/topmentions/TopMentionsService.scala

```diff
@@ -0,0 +1,38 @@
+package topmentions
+
+import common.{Box, GuLogging}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class TopMentionsService(topMentionsS3Client: TopMentionsS3Client) extends GuLogging {
+
+  private val topMentions = Box[Option[Map[String, TopMentionsDetails]]](None)
+
+  def refreshTopMentions()(implicit executionContext: ExecutionContext): Future[Unit] = {
+    val retrievedTopMentions = topMentionsS3Client.getListOfKeys().map { key => key.map { retrieveTopMention(_) } }
```

Comment:
do we just get everything from s3? Is there an aws sdk "getall" type of function we could use?

Reply:
Makes sense 👍